From 5581cf563bfdcc9068268dcaf3027b41acf4a333 Mon Sep 17 00:00:00 2001
From: Cici Ruan <cuicuiruan@google.com>
Date: Wed, 10 May 2023 20:09:53 -0700
Subject: [PATCH] Restore usage of base::Value::DictSize

This restores base::Value::DictSize for the packages which are still
using the deprecated function.

Change-Id: I29bb087391e80e828235d921a93943cdbeffadc5
---
 base/values.cc | 4 ++++
 base/values.h  | 3 +++
 2 files changed, 7 insertions(+)

diff --git a/base/values.cc b/base/values.cc
index 49b43a07b4..3257cb8b48 100644
--- a/base/values.cc
+++ b/base/values.cc
@@ -1204,6 +1204,10 @@ Value* Value::SetDoubleKey(StringPiece key, double value) {
   return GetDict().Set(key, value);
 }
 
+size_t Value::DictSize() const {
+  return GetDict().size();
+}
+
 absl::optional<bool> Value::FindBoolPath(StringPiece path) const {
   return GetDict().FindBoolByDottedPath(path);
 }
diff --git a/base/values.h b/base/values.h
index d168df3c17..3de7729152 100644
--- a/base/values.h
+++ b/base/values.h
@@ -819,6 +819,9 @@ class BASE_EXPORT GSL_OWNER Value {
 
   // Note: Do not add more types. See the file-level comment above for why.
 
+  // DEPRECATED: prefer `Value::Dict::size()`.
+  size_t DictSize() const;
+
   // Comparison operators so that Values can easily be used with standard
   // library algorithms and associative containers.
   BASE_EXPORT friend bool operator==(const Value& lhs, const Value& rhs);
-- 
2.39.2

